Master files to be used for view overrides

schnuti opened the pull request:
Hi there,
I added those 3 files to DPfields.
e.g. 
1. Copy the article view default.php to /your/template/html/com_content/article/default.php
2. Edit that file and add code as in the example file. Be aware of the fieldlist. You should add all fields you have available for articles with id.title. Order them as you like. Set the value to true if it should be shown here, false iif not. You can duplicate it for other positions in the default.php
3. Copy the erender.php file to  /your/template/html/layouts/com_content/field/erender.php You should be ok leaving this as is for now. You can chane it if you need othe html tags.
4. Copy efield1.php to  /your/template/html/layouts/com_content/field/erender/efield1.php Change the name of this file to efield# whwrw # is the id of a field you want to test giving a specific JLayout.

Nice?

schnuti committed:
Output for overrides 1

schnuti committed:
Files for override 2

schnuti committed:
Output for overrides 3

laoneo commented:
Would it not make more sense to add a layout per field type like sql.php? If a user then wants to add field independent code he can add some conditions in his override like:

```
if (! key_exists('field', $displayData))
{
    return;
}

$field = $displayData['field'];
if (! $field)
{
    return;
}

if ($field->id == 3)
{
    echo 'foo';
}
else if ($field->id == 4)
{
    echo 'bar';
}
?>
```

The problem I see with your request is that there is a lot of logic in the layouts from a none PHP developer point of view. A layout should be as small as possible. The admin has then to change only in one field if he wants to change some output.

schnuti commented:
What do you mean by field type?
If I want to change something for 1 (one) field, it mostly has with html and html-tags to do. It do not mainly depend on field type. <li>, <dd>, <div>, <span> ......

In your example you have a lot of code the user must change. I have first none. Except in the default.php 
Place the code where the fields are to appear. Yes, he has to add the fields to the array. Thats it.
If he wants to change it further, he edits the few rows at the bottom of the JLayout file. If he wants to explicit change 1 field he adds and edits the few lines at the bottom of the efieldx,php.

I hate the argument - No php, the funny user do not understand it - 
I agree, it would be nice but you can only have it with flexibility = 0 or you have to live with those if's and foreach's.
As I wrote: the article default.php is an example. What would the user say if that was written in static code?

If you really want to code, then you add all the fields by normal php echo in the default.php as everything as done before JLayout.

What I do not know is how heavy the JLayout files are. Not only every single file has to be found and loaded into memory, there is also a long look up sequence for finding them in those predefind directories.

If they are lightweight, I would use them a lot for single fields. I now use them for reusable areas like header and footer, or where I can use foreach for a bunch of fields.

schnuti committed:
Added load JFile

Managing fields without plugin enabled.
As far as I can see, the plugin is not used here so why not let 'em play with the fields ith disabld plugin.

schnuti committed:
Correction

Correction

schnuti committed:
Added load JFile 

Managing fields without plugin enabled.
As far as I can see, the plugin is not used here so why not let 'em play with the fields ith disabld plugin.

schnuti committed:
Merge pull request #1 from schnuti/schnuti-jfile1

Added load JFile
